### PR TITLE
[CODE UNIFICATION] Introduce protocols for Ad Unit Configuration

### PR DESCRIPTION
**Pull request by OlenaPostindustria**

### CODE REVIEW

**Review by ValentinPostindustria — APPROVED** [review]
LGTM

File: Example/PrebidDemo/PrebidDemoTests/PrebidDemoTests.swift [hunk 1]
```diff
@@ -274,7 +274,7 @@ class PrebidDemoTests: XCTestCase {
             if first != -1 {
                 first = -1
 
-                prebidbBannerAdUnit.adSizes = [CGSize(width: 728, height: 90)]
+                prebidbBannerAdUnit.adUnitConfig.adSize = CGSize(width: 728, height: 90)
```
Comment (YuriyVelichkoPI) [inline Example/PrebidDemo/PrebidDemoTests/PrebidDemoTests.swift:277]: we should keep the original API here

File: InternalTestApp/PrebidMobileDemoRendering/ViewControllers/Adapters/Prebid/GAM/PrebidGAMNativeAdController.swift [hunk 2]
```diff
@@ -143,7 +143,13 @@ class PrebidGAMNativeAdController: NSObject, AdaptedController {
     // MARK: - Helpers
     
     private func setupNativeAdUnit() {
-        adUnit = NativeRequest(configId: prebidConfigId, assets: nativeAssets ?? [], eventTrackers: eventTrackers ?? [])
+        adUnit = NativeRequest(configId: prebidConfigId)
```
Comment (YuriyVelichkoPI) [inline InternalTestApp/PrebidMobileDemoRendering/ViewControllers/Adapters/Prebid/GAM/PrebidGAMNativeAdController.swift:146]: we should keep API here

File: InternalTestApp/PrebidMobileDemoRendering/ViewControllers/Adapters/Prebid/OpenX/PrebidNativeAdController.swift [hunk 3]
```diff
@@ -148,7 +148,13 @@ class PrebidNativeAdController: NSObject, AdaptedController {
     // MARK: - Helpers
     
     private func setupNativeAdUnit(configId: String) {
-        adUnit = NativeRequest(configId: configId, assets: nativeAssets ?? [], eventTrackers: eventTrackers ?? [])
+        adUnit = NativeRequest(configId: configId)
```
Comment (YuriyVelichkoPI) [inline InternalTestApp/PrebidMobileDemoRendering/ViewControllers/Adapters/Prebid/OpenX/PrebidNativeAdController.swift:151]: we should keep API here

Comment (YuriyVelichkoPI) [inline InternalTestApp/PrebidMobileDemoRendering/ViewControllers/Adapters/Prebid/OpenX/PrebidNativeAdController.swift:151]: we should keep API here

File: InternalTestApp/PrebidMobileDemoRendering/ViewControllers/Adapters/Prebid/OpenX/PrebidNativeAdFeedController.swift [hunk 4]
```diff
@@ -147,7 +147,15 @@ class PrebidNativeAdFeedController: NSObject, PrebidConfigurableController {
     // MARK: - Helpers
     
     private func setupNativeAdUnit(configId: String) {
-        adUnit = NativeRequest(configId: configId, assets: nativeAssets ?? [], eventTrackers: eventTrackers ?? [])
+        adUnit = NativeRequest(configId: configId)
```
Comment (YuriyVelichkoPI) [inline InternalTestApp/PrebidMobileDemoRendering/ViewControllers/Adapters/Prebid/OpenX/PrebidNativeAdFeedController.swift:150]: we should keep API here

File: InternalTestApp/PrebidMobileDemoRendering/ViewControllers/TableCells/FeedGAMAdTableViewCell.swift [hunk 5]
```diff
@@ -69,7 +69,15 @@ class FeedGAMAdTableViewCell: UITableViewCell {
     // MARK: - Helpers
     
     private func setupNativeAdUnit(configId: String) {
-        adUnit = NativeRequest(configId: configId, assets: nativeAssets ?? [], eventTrackers: eventTrackers ?? [])
+        adUnit = NativeRequest(configId: configId)
```
Comment (YuriyVelichkoPI) [inline InternalTestApp/PrebidMobileDemoRendering/ViewControllers/TableCells/FeedGAMAdTableViewCell.swift:72]: we should keep API here

File: PrebidMobile/AdUnits/AdUnit.swift [hunk 6]
```diff
@@ -74,15 +69,6 @@ import ObjectiveC.runtime
     //TODO: dynamic is used by tests
     dynamic public func fetchDemand(adObject: AnyObject, completion: @escaping(_ result: ResultCode) -> Void) {
         
-        if !(self is NativeRequest){
```
Comment (YuriyVelichkoPI) [inline PrebidMobile/AdUnits/AdUnit.swift:77]: Are you sure that we can remove this code?

File: PrebidMobileTests/AdUnitTests/AdUnitSuccessorObjCTests.m [hunk 7]
```diff
@@ -73,33 +73,33 @@ - (void)testBannerParametersCreation {
...
-    AdUnit *adunit = [[InterstitialAdUnit alloc] initWithConfigId:configId minWidthPerc:50 minHeightPerc:70];
+    BaseAdUnit *adunit = [[InterstitialAdUnit alloc] initWithConfigId:configId minWidthPerc:50 minHeightPerc:70];
     XCTAssertNotNil(adunit);
 }
 
 //MARK: - VideoAdUnit
 - (void)testVideoAdUnitCreation {
     //when
-    AdUnit *adunit = [[VideoAdUnit alloc] initWithConfigId:configId size:CGSizeMake(300, 250)];
+    BaseAdUnit *adunit = [[VideoAdUnit alloc] initWithConfigId:configId size:CGSizeMake(300, 250)];
```
Comment (YuriyVelichkoPI) [inline PrebidMobileTests/AdUnitTests/AdUnitSuccessorObjCTests.m:88]: Why we can't use AdUnit anymore?

File: PrebidMobileTests/AdUnitTests/AdUnitSuccessorTests.swift [hunk 8]
```diff
@@ -39,7 +39,8 @@ class AdUnitSuccessorTests: XCTestCase {
 
     func testBannerAdUnitAddSize() {
         let adUnit = BannerAdUnit(configId: Constants.configID1, size: CGSize(width: Constants.width1, height: Constants.height1))
-        adUnit.adSizes = [CGSize(width: Constants.width1, height: Constants.height1), CGSize(width: Constants.width2, height: Constants.height2)]
+        adUnit.adUnitConfig.adSize = CGSize(width: Constants.width2, height: Constants.height2)
```
Comment (YuriyVelichkoPI) [inline PrebidMobileTests/AdUnitTests/AdUnitSuccessorTests.swift:42]: we should keep API here

File: PrebidMobileTests/AdUnitTests/AdUnitTests.swift [hunk 9]
```diff
@@ -36,7 +34,7 @@ class AdUnitTests: XCTestCase {
         var resultCode: ResultCode?
         
         let expected = ResultCode.prebidDemandFetchSuccess
-        let adUnit = AdUnit(configId: "138c4d03-0efb-4498-9dc6-cb5a9acb2ea4", size: CGSize(width: 300, height: 250))
```
Comment (YuriyVelichkoPI) [inline PrebidMobileTests/AdUnitTests/AdUnitTests.swift:39]: Need to be sure that we still have the original construction method. I saw that it was removed above.

**Review by YuriyVelichkoPI — CHANGES_REQUESTED** [review]

**Review by YuriyVelichkoPI — APPROVED** [review]
LGTM

**Pull request merged**